### PR TITLE
remove MkFit clang16 vectorization warnings

### PR DESCRIPTION
**Pull request by dan131riley**
#### PR description:

Removes the "loop not vectorized" warnings in https://github.com/cms-sw/cmssw/issues/41624.  Four of the loops are not vectorized by any of our compilers, so I commented out the simd pragma for those.  For the others that are vectorized by icc, I added an ifdef to disable the simd pragma for clang.  I would have preferred to suppress the warning messages with a compiler pragma, but that doesn't seem to be possible for the case where vectorization is supposed to be forced.

#### PR validation:

Compiles without warnings with clang16 and gcc.

### CODE REVIEW

File: RecoTracker/MkFitCore/src/MkFinder.cc [hunk 1]
```diff
@@ -716,7 +720,7 @@ namespace mkfit {
 
       mhp.reset();
 
-#pragma omp simd
+      //#pragma omp simd doesn't vectorize with current compilers
```
Comment (slava77) [inline RecoTracker/MkFitCore/src/MkFinder.cc:723]: is the indentation coming from code-check/code-format?
Having no indent as in the original pragma seems better.

**Review by slava77 — COMMENTED** [review]
I think that some comments are needed inline to explain `#if !defined(__clang__)`

File: RecoTracker/MkFitCore/src/MkFinder.cc [hunk 2]
```diff
@@ -300,7 +300,9 @@ namespace mkfit {
 
     if (L.is_barrel()) {
       // Pull out the part of the loop that vectorizes
+#if !defined(__clang__)
```
[the remark this answers is not in the document]
Comment (dan131riley) [reply RecoTracker/MkFitCore/src/MkFinder.cc:307]: Comments added

[the remark this answers is not in the document]
Comment (slava77) [reply RecoTracker/MkFitCore/src/MkFinder.cc:307]: > Comments added

Thank you.
I think this can go to the upstream/cms-sw

**Pull request merged**